gcc/rust/typecheck/rust-hir-type-check-pattern.cc

```diff
@@ -43,32 +43,97 @@ TypeCheckPattern::Resolve (HIR::Pattern *pattern, TyTy::BaseType *parent)
 void
 TypeCheckPattern::visit (HIR::PathInExpression &pattern)
 {
-  infered = TypeCheckExpr::Resolve (&pattern);
-
-  /*
-   * We are compiling a PathInExpression, which can't be a Struct or Tuple
-   * pattern. We should check that the declaration we are referencing IS NOT a
-   * struct pattern or tuple with values.
-   */
+  // Pattern must be enum variants, sturcts, constants, or associated constansts
+  TyTy::BaseType *pattern_ty = TypeCheckExpr::Resolve (&pattern);
+
+  NodeId ref_node_id = UNKNOWN_NODEID;
+  bool maybe_item = false;
+  maybe_item
+    |= resolver->lookup_resolved_name (pattern.get_mappings ().get_nodeid (),
+				       &ref_node_id);
+  maybe_item
+    |= resolver->lookup_resolved_type (pattern.get_mappings ().get_nodeid (),
+				       &ref_node_id);
```

Comment:
Why is there the same line twice? Am I missing something?

Reply:
First one checks variable names and second one checks type names. E.g. first for static items, second for type aliases

Reply:
I'll leave this one to the judgement of the code maintainers. I'm not familiar with how this code works

Reply:
Maybe you dont recognize different methods are called

Reply:
oh lol. I'm sorry. I spent so long reading that code and didn't notice